android-json-form-wizard/src/main/java/com/vijay/jsonwizard/rules/RulesEngineFactory.java

```diff
@@ -257,19 +257,31 @@ public void afterEvaluate(Rule rule, Facts facts, boolean evaluationResult) {
         //Overriden
     }
 
+
     @Override
     public void beforeExecute(Rule rule, Facts facts) {
-        //Overriden
+        Timber.e("Putting facts in beforeExecute");
+        HashMap<String, Object> myMap = new HashMap<>();
+        facts.put("facts", myMap);
     }
 
     @Override
     public void onSuccess(Rule rule, Facts facts) {
-        //Overriden
+        Timber.e("Putting facts in onSuccess");
+        HashMap<String, Object> myMap = facts.get("facts");
+
+        for (String key :
+                myMap.keySet()) {
+            facts.put(key, myMap.get(key));
+        }
+
+        facts.remove("facts");
     }
 
     @Override
     public void onFailure(Rule rule, Facts facts, Exception exception) {
-        //Overriden
+        Timber.e("Putting facts in onFailure");
```

Comment:
Kindly check if there might be successful executions of rules meaning that we might need to update our `facts` object with `key-values` from the `facts.facts` map